execution_engine/src/core/tracking_copy/mod.rs

```diff
@@ -231,6 +266,29 @@ impl<R: StateReader<Key, StoredValue>> TrackingCopy<R> {
         }
     }
 
+    pub fn get_keys(
+        &mut self,
+        correlation_id: CorrelationId,
+        key_tag: &KeyTag,
+    ) -> Result<BTreeSet<Key>, R::Error> {
+        let mut ret: BTreeSet<Key> = BTreeSet::new();
+        match self.cache.get_key_tag_reads_cached(&key_tag) {
+            Some(keys) => ret.extend(keys),
+            None => {
+                let key_tag = key_tag.to_owned();
+                let keys = self
+                    .reader
+                    .keys_with_prefix(correlation_id, &[key_tag as u8])?;
+                ret.extend(keys);
+                self.cache.insert_key_tag_read(key_tag, ret.to_owned())
+            }
+        }
```

Comment:
You could do something like:

```rust
        // Do a cached read of the keys under a tag
        let key_tag_reads = match self.cache.get_key_tag_reads_cached(&key_tag) {
            Some(keys) => keys,
            None => {
                let key_tag = key_tag.to_owned();
                let keys = self
                    .reader
                    .keys_with_prefix(correlation_id, &[key_tag as u8])?;
                self.cache.insert_key_tag_read(key_tag, keys.to_owned())
                keys
            }
        }
        ret.extend(key_tag_reads);
```

Reply:
Good one, will do.

Reply:
Actually some impediments:
- currently `StateReader::keys_with_prefix` returns a `Vec` of keys.  I could change this to a `BTreeSet`, but...
- there's an ownership problem: `TrackingCopyCache::get_key_tags_reads_cached` returns a reference to the set but `StateReader::keys_with_prefix` returns the set itself.  The latter is unavoidable, but I don't want to have to clone if we change the former.

I may still change `StateReader::keys_with_prefix` to return a `BTreeSet` for consistency, but will likely do in a follow-up PR.

Reply:
Sounds good.